### PR TITLE
Home_Work_5

### PR DESCRIPTION
**Pull request by Kantsum**
№4

### CODE REVIEW

**Comment by Kantsum** [comment]
14

File: Tasks/Kantsumer/HW3.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                for n in range(len(z)):
+                    if n == len(z) - 1:
+                        new_words += z[n]
+                    else:
+                        new_words += z[n] + ' ' + ' ' * (a // b) + ' ' * (r > 0)
+                    r -= 1
+
+            new_text.write(new_words + '\n')
+
+            new_words = new_words[x:]
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW3.py:31]: nice, this is the optimal solution for such task

File: Tasks/Kantsumer/HW4.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ for i in range(1,len(l)):
+      if l[i]-1 != l[i-1] :
+        new_l.append(str(c1+c2))
+        c1=str(l[i])
+        c2=''
+      else:
+          c2='-'+str(l[i])
+ else:
+     new_l.append(str(c1+c2))    
+ return(new_l)
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW4.py:28]: indention???

File: Tasks/Kantsumer/HW4b.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        l = [int(i) for i in l]
+        break
+    except ValueError:
+        print('Wrong input')
+    # l=[28,5,39,11,44,3,66,5,77]#List pay visitoe
+ return(l)
+
+def get_bill(l:list):
+ 
+ sv=len(l)
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW4b.py:16]: indention once again, maybe problem with your IDE

File: Tasks/Kantsumer/HW5.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+print(sorted(list(filter(lambda x : x%2!=0, map(int,['1', '11', '12', '22', '2', '13', '30', '33'])))))
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW5.py:1]: **key** arg is missing in the sorted function

File: Tasks/Kantsumer/HW5.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+l=[1, 2, [2, 4, [[7, 8], 4, 6]]]
+# # l=[1, 2, 6,7]
+
+def sum_l(l):
+    all = 0
+    for i in l:
+        if isinstance(i,list):
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW5.py:12]: nice

File: Tasks/Kantsumer/HW5.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+def fib(n):
+    i=0
+    a=1
+    b=1
+    if n==1:
+     print(0, end='')
+    elif n==2:
+     print(0,',',1,end='')
+    elif n>=3:
+        print(0,',',1,end=' ')
```
Comment (MikitaTsiarentsyeu) [inline Tasks/Kantsumer/HW5.py:33]: clever logic for displaying

**Review by MikitaTsiarentsyeu — APPROVED** [review]
very good, grant you 9 for each task

**Pull request merged**